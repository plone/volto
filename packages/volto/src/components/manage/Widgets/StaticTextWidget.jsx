import PropTypes from 'prop-types';

import FormFieldWrapper from '@plone/volto/components/manage/Widgets/FormFieldWrapper';

const StaticTextWidget = (props) => {
  const { id, value } = props;

  return (
    <FormFieldWrapper {...props} className="text" columns={1}>
      <div id={id} className="wrapper">
        <p
          dangerouslySetInnerHTML={{
            __html: value?.data || '',
          }}
        />
      </div>
    </FormFieldWrapper>
  );
};

export default StaticTextWidget;

StaticTextWidget.propTypes = {
  id: PropTypes.string.isRequired,
  value: PropTypes.string,
  minLength: PropTypes.number,
  maxLength: PropTypes.number,
};

StaticTextWidget.defaultProps = {
  value: null,
  minLength: null,
  maxLength: null,
};
