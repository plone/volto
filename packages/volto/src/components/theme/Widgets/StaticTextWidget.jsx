import React from 'react';
import cx from 'classnames';

const StaticTextWidget = ({ value, className }) =>
  value ? (
    <p
      className={cx(className, 'statictext', 'widget')}
      dangerouslySetInnerHTML={{
        __html: value.data,
      }}
    />
  ) : (
    ''
  );

export default StaticTextWidget;
