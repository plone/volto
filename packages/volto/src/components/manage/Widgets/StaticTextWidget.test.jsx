import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import StaticTextWidget from './StaticTextWidget';

const mockStore = configureStore();

test('renders a text widget component', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });

  const component = renderer.create(
    <Provider store={store}>
      <StaticTextWidget id="my-field" />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
