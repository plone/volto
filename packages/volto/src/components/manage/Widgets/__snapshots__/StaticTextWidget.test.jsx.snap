// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a text widget component 1`] = `
<div
  className="inline field text field-wrapper-my-field"
>
  <div
    className="ui grid"
  >
    <div
      className="stretched row"
    >
      <div
        className="twelve wide column"
      >
        <div
          className="wrapper"
          id="my-field"
        >
          <p
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
